src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionInfo.cs

```diff
@@ -99,6 +99,15 @@ public ICollection<string> AllConnectionTypes
             }
         }
 
+        public bool HasConnectionType(string connectionType)
+        {
+            if (connectionType == null)
+            {
+                connectionType = ConnectionType.Default;
+            }
+            return ConnectionTypeToConnectionMap.ContainsKey(connectionType);

```

Comment:
Is this a ConcurrentDictionary we're operating on? If so, I'd highly recommend that getting a specific connection type be done via a TryGet instead of a ContainsKey and Get that way we perform the check and get in an atomic operation.